src/main/java/com/ldtteam/structurize/items/ItemCommandTool.java

```diff
@@ -0,0 +1,464 @@
+        if (reader.canRead() && reader.peek() == '/') { reader.read(); }
+
+        final CommandDispatcher<CommandSourceStack> dispatcher = ((ServerLevel)context.getLevel()).getServer().getCommands().getDispatcher();
+        final ParseResults<CommandSourceStack> parsed = dispatcher.parse(reader, command.getCommandBlock().createCommandSourceStack());
+        if (parsed.getReader().canRead() || parsed.getContext().getNodes().size() < 4
+                || !parsed.getContext().getNodes().get(0).getNode().getName().equals(MOD_ID)
+                || !parsed.getContext().getNodes().get(1).getNode().getName().equals(ScanCommand.NAME))
+        {
+            player.sendMessage(new TranslatableComponent("com.ldtteam.structurize.gui.commandtool.copycommand.notscan"), player.getUUID());
+            return InteractionResult.FAIL;
+        }
+
+        final CommandContext<CommandSourceStack> cmdContext = parsed.getContext().build(parsed.getReader().getString());
+        try
+        {
+            final BlockPos from = BlockPosArgument.getSpawnablePos(cmdContext, ScanCommand.POS1);
+            final BlockPos to = BlockPosArgument.getSpawnablePos(cmdContext, ScanCommand.POS2);
+            BlockPos anchor = null;
+            if (parsed.getContext().getArguments().containsKey(ScanCommand.ANCHOR_POS))
+            {
+                anchor = BlockPosArgument.getSpawnablePos(cmdContext, ScanCommand.ANCHOR_POS);
+            }
+            String name = "";
+            if (parsed.getContext().getArguments().containsKey(ScanCommand.FILE_NAME))
+            {
+                name = StringArgumentType.getString(cmdContext, ScanCommand.FILE_NAME);
+            }
+
+            setData(context.getItemInHand(), context.getLevel(), pos, getTeleportPos(from, to), name);
+
+            ItemScanTool.setBounds(scanTool, from, to);
+            ItemScanTool.setAnchorPos(scanTool, anchor);
+            Network.getNetwork().sendToPlayer(new ShowScanMessage(from, to, anchor, name), (ServerPlayer) player);
+
+            player.sendMessage(new TranslatableComponent("com.ldtteam.structurize.gui.commandtool.copycommand.ok", name), player.getUUID());
+            player.playNotifySound(SoundEvents.NOTE_BLOCK_CHIME, SoundSource.PLAYERS, 1.0F, 1.0F);
+            return InteractionResult.SUCCESS;
+        }
+        catch (CommandSyntaxException e)
+        {
+            player.sendMessage(new TranslatableComponent("com.ldtteam.structurize.gui.commandtool.copycommand.notscan"), player.getUUID());
+            return InteractionResult.FAIL;
+        }
+    }
+
+    /**
+     * Searches the player's hotbar (only) for a scan tool.
+     * If there's multiple, it prefers the closest item to the current slot, starting from the left side.
+     * @param player The player to search.
+     * @return The scan tool found, or an empty stack.
+     */
+    @NotNull
+    private ItemStack findScanTool(@NotNull final Player player)

```

Comment:
There is similar logic in mcol for searching any item in inv, the double for is not needed iirc (same behaviour is provided by implementation of player inv iirc)

Reply:
Code in mcol is not helpful, since the dependency points the other way.

The double for is entirely necessary (as is not using other code) because this implements a very specific search order to find the "correct" scan tool, in case someone is carrying more than one.  This is documented.